pkg/registry/apis/provisioning/secrets/secret.go

```diff
@@ -34,12 +35,14 @@ var _ Service = (*secretsService)(nil)
 type secretsService struct {
 	secureValues SecureValueClient
 	decryptSvc   grafanasecrets.DecryptService
+	cfg          *setting.Cfg
 }
 
-func NewSecretsService(secretsSvc SecureValueClient, decryptSvc grafanasecrets.DecryptService) Service {
+func NewSecretsService(secretsSvc SecureValueClient, decryptSvc grafanasecrets.DecryptService, cfg *setting.Cfg) Service {
```

Comment:
I would not pass the entire config but just a single param for one or more decrypters.